xml/map.go

```diff
@@ -0,0 +1,61 @@
+package xml
+
+// mapEntryWrapper is the default member wrapper tag name for XML Map type
+const mapEntryWrapper = "entry"
+
+// Map represents the encoding of a XML map type
+type Map struct {
+	w       writer
+	scratch *[]byte
+
+	memberWrapperName  string
+	memberStartElement *StartElement
+	memberEndElement   *EndElement
+
+	mapEndElement *EndElement
+}
```

Comment:
Would be nice to have documentation for these element members. Also highlighting their usage flatten vs unflatten